### DISCUSSION TITLE
C# StepwisePlanner How to ask for more information and confirmation from user

### DISCUSSION THREAD

**Discussion by hooyao** [microsoft/semantic-kernel#2213] [Q&A]
I'm using `StepwisePlanner` in a business chatbot, with Azure Openai `text-davinci-003`. Our team net yet has access to GPT4 yet. And at the moment, gpt-35-turbo/gpt-35-turbo-16k are slow and less performant.
We have a skill to compute the estimated time of running a job. It's like 

```
    [SKFunction,
     Description("Compute the estimated time in seconds for the Google drive migration for a given number of items. ")]
    public long GetGoogleDriveEstimation(
        [Description("The folder and file count that need to be migrated to Google Drive")]
        int itemCount)
```

## When the prompt is as follows, the user doesn't provide the required input
```
Context: If the user doesn't provide the required information for the required skill, please ask the user to input required information instead of using any default value. User question: I'm migrating files from Google Drive to OneDrive. How long will it take?
```
Though it the result is acceptable ``, but the planner still try to use the empty argument on the skill function.
```
[ACTION] {"action": "MigrationEstimationSkill.GetGoogleDriverEstimation","action_variables": {"input":"","itemCount":""}}
[OBSERVATION] Error occurred: System.ArgumentOutOfRangeException:  is not a valid value for Int32. (Parameter 'value') (Parameter 'itemCount')
      Actual value was .
         at Microsoft.SemanticKernel.SkillDefinition.SKFunction.<>c__DisplayClass43_1.<GetParameterMarshalerDelegate>g__Process|5(String value, <>c__DisplayClass43_2&) in E:\Projects\ServiceCopilot\semantickernel\semantic
-kernel\dotnet\src\SemanticKernel\SkillDefinition\SKFunction.cs:line 619
```
What shall I do to let planner understand it shouldn't execute the function because  `"itemCount":""` is empty?

## What if GetGoogleDriveEstimation is a write operation which need user's confirmation? 
I tried to add following reminding to the prompt, but it doesn't work
```
   [SKFunction,
     Description("Compute the estimated time in seconds for the Google drive migration of a given number of items. " +
                 "This is a dangerous operation. Please ask for usr's confirmation before proceeding.")]
and "For dangerous operation, please ask for user's confirmation before using the skill." to the planner prompt.
```
The planner still ignores the confirmation request and executes the skill.
 
How can I let the planner first ask for user's confirmation before executing this skill function?

**Comment by lemillermicrosoft** [comment]
Do you have a skill available that is for getting user input/confirmation? Could you share all Skills (and functions) that are available? Is it possible to make the user confirmation a part of the skill function itself so it's done automatically each time?
